### PR TITLE
feat(delete): Add delete by

### PR DESCRIPTION
**Pull request by ElPicador**

### CODE REVIEW

File: src/main/java/com/algolia/search/saas/Index.java [hunk 1]
```diff
@@ -586,24 +586,55 @@ public JSONObject deleteObject(String objectID, RequestOptions requestOptions) t
...
+   * @param query the query string
+   */
+  @Deprecated
   public void deleteByQuery(Query query) throws AlgoliaException {
     deleteByQuery(query, 100000);
   }
 
   /**
    * Delete all objects matching a query
+   * Deprecated use deleteBy
```
Comment (clement-leprovost) [inline src/main/java/com/algolia/search/saas/Index.java]: Can we use a Javadoc `@deprecated` tag here?

File: src/main/java/com/algolia/search/saas/Index.java [hunk 2]
```diff
@@ -586,24 +586,55 @@ public JSONObject deleteObject(String objectID, RequestOptions requestOptions) t
...
   public void deleteByQuery(Query query, RequestOptions requestOptions) throws AlgoliaException {
     this.deleteByQuery(query, 100000, requestOptions);
   }
 
-  public void deleteByQuery(Query query, int batchLimit) throws AlgoliaException {
-    this.deleteByQuery(query, batchLimit, RequestOptions.empty);
-  }
-
+  /**
+   * Deprecated use deleteBy
```
Comment (clement-leprovost) [inline src/main/java/com/algolia/search/saas/Index.java]: Same

File: src/test/java/com/algolia/search/saas/SimpleTest.java [hunk 3]
```diff
@@ -182,6 +183,18 @@ public void deleteByQuery() throws JSONException, AlgoliaException {
   }
 
   @Test
+  public void deleteBy() throws JSONException, AlgoliaException {
+    JSONObject task = index.addObjects(new JSONArray().put(new JSONObject()
+      .put("name", "Washington"))
+      .put(new JSONObject().put("name", "San Francisco"))
+      .put(new JSONObject().put("name", "San Jose")));
+    index.waitTask(task.getString("taskID"));
+    index.deleteBy(new Query("San"));
```
Comment (clement-leprovost) [inline src/test/java/com/algolia/search/saas/SimpleTest.java]: You should wait on the task; delete by query is asynchronous like all write operations.

File: src/main/java/com/algolia/search/saas/Index.java (hunk 1 above)
Comment (ElPicador) [reply src/main/java/com/algolia/search/saas/Index.java]: oh there is one? Nice

File: src/test/java/com/algolia/search/saas/SimpleTest.java (hunk 3 above)
Comment (ElPicador) [reply src/test/java/com/algolia/search/saas/SimpleTest.java]: nice catch

**Review by clement-leprovost — APPROVED** [review]
LGTM 👍

**Pull request merged**